src/test/java/fi/otavanopisto/kuntaapi/test/server/integration/management/FragmentTestsIT.java

```diff
@@ -0,0 +1,147 @@
+    String organizationId = getOrganizationId(0);
+    given() 
+      .baseUri(getApiBasePath())
+      .contentType(ContentType.JSON)
+      .get("/organizations/{organizationId}/fragments/{fragmentId}", organizationId, getOrganizationFragmentId(organizationId, 0))
+      .then()
+      .assertThat()
+      .statusCode(200)
+      .body("id", notNullValue())
+      .body("slug", is("test-0"))
+      .body("contents", containsString("consectetur adipiscing elit"));
+  } 
+  
+  @Test
+  public void testListFragmentsBySlug() {
+    given() 
+      .baseUri(getApiBasePath())
+      .contentType(ContentType.JSON)
+      .get("/organizations/{organizationId}/fragments?slug=test-1", getOrganizationId(0))
+      .then()
+      .assertThat()
+      .statusCode(200)
+      .body("id.size()", is(1))
+      .body("id[0]", notNullValue())
+      .body("slug[0]", is("test-1"));
+    
+    given() 
+      .baseUri(getApiBasePath())
+      .contentType(ContentType.JSON)
+      .get("/organizations/{organizationId}/fragments?slug=non-existing", getOrganizationId(0))
+      .then()
+      .assertThat()
+      .statusCode(200)
+      .body("id.size()", is(0));
+  } 
+  
+  @Test
+  public void testListFragments() {
+    given() 
+      .baseUri(getApiBasePath())
+      .contentType(ContentType.JSON)
+      .get("/organizations/{organizationId}/fragments", getOrganizationId(0))
+      .then()
+      .assertThat()
+      .statusCode(200)
+      .body("id.size()", is(3))
+      .body("id[1]", notNullValue())
+      .body("slug[1]", is("test-1"))
+      .body("contents[1]", containsString("Another test fragment"));
+  } 
+  
+  @Test
+  public void testOrganizationFragmentsNotFound() throws InterruptedException {
+    String organizationId = getOrganizationId(0);
+    String incorrectOrganizationId = getOrganizationId(1);
+    String organizationAnnouncecmentId = getOrganizationFragmentId(organizationId, 0);
+    
+    String[] malformedIds = new String[] {"evil", "*", "/", "1", "-1", "~"};
+    assertFound(String.format("/organizations/%s/fragments/%s", organizationId, organizationAnnouncecmentId));
+    assertEquals(3, countApiList(String.format("/organizations/%s/fragments", organizationId)));
+    
+    for (String malformedId : malformedIds) {
+      assertNotFound(String.format("/organizations/%s/fragments/%s", organizationId, malformedId));
+    }
+    
+    assertNotFound(String.format("/organizations/%s/fragments/%s", incorrectOrganizationId, organizationAnnouncecmentId));
+    assertEquals(0, countApiList(String.format("/organizations/%s/fragments", incorrectOrganizationId)));
+  }
+
+  private void createPtvSettings() {
+    insertSystemSetting(PtvConsts.SYSTEM_SETTING_BASEURL, String.format("%s%s", getWireMockBasePath(), BASE_URL));
+    flushCache();
+  }
+  
+  private void deletePtvSettings() {
+    deleteSystemSetting(PtvConsts.SYSTEM_SETTING_BASEURL);
+  }
+    
+  private void createManagementSettings(String organizationId) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "createManagementSettings" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)